README.md

```diff
@@ -1,21 +1,332 @@
+This library implements only enough of the TensorFlow C API to allow inference, no training.
+
+Project status: This project is currently in pre-release with documentation and code being prepared for a first release.
+As such breaking changes may be made.
+If you are interested in using this library please get in touch.
+
+
+## Installation
+
+### Dependencies
+
+To install the library requires the following to be installed on the system:
+
+* cmake >= 3.1
+* TensorFlow C API, download from <https://www.tensorflow.org/install/lang_c><sup>1</sup>
+* Fortran and C compilers
+
+<sup>1</sup> Note that this page sometimes does not list the latest version of
+the library.  You can try altering the library download URLs on the page to
+reflect the newest version.  E.g. if the URL ends `...-2.11.tar.gz` try
+changing it to `...-2.13.tar.gz`.
+
+### Library installation
+
+To build and install the library:
+
+1. Navigate to the location in which you wish to install the source and run:  
+    ```
+    git clone git@github.com:Cambridge-ICCS/fortran-tf-lib.git
+    ```
+    to clone via ssh, or  
+    ```
+    git clone https://github.com/Cambridge-ICCS/fortran-tf-lib.git
+    ```
+    to clone via https.  
+2. Navigate into the library directory by running:  
+    ```
+    cd fortran-tf-lib/fortran-tf-lib/
+    ```
+3. Create a `build` directory and execute cmake from within it using the relevant flags:  
+    ```
+    mkdir build
+    cd build
+    cmake .. -DCMAKE_BUILD_TYPE=Release
+    ```
+    It is likely that you will need to provide at least the `TENSORFLOW_LOCATION` flag.  
+    The Fortran compiler must be the same one that you are planning to compile your Fortran
+    code with.  It is advisable to use C and Fortran compilers from the same provider.
+
+    The following CMake flags are available and can be passed as arguments through `-D<Option>=<Value>`:
+    | Option                                                                                            | Value                        | Description                                                   |
+    | ------------------------------------------------------------------------------------------------- | ---------------------------- | --------------------------------------------------------------|
+    | [`CMAKE_Fortran_COMPILER`](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER.html) | `ifort` / `gfortran`         | Specify a Fortran compiler to build the library with. This should match the Fortran compiler you're using to build the code you are calling this library from.        |
+    | [`CMAKE_C_COMPILER`](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER.html)       | `icc` / `gcc`                | Specify a C compiler to build the library with.                |
+    | `TENSORFLOW_LOCATION`<sup>2</sup>   | `</path/to/tensorflow/>`          | Location of TensorFlow C API installation<sup>1</sup>. |
```

Comment:
`IMPORTED_LOCATION` as opposed to `TENSORFLOW_DIR` or `CMAKE_PREFIX_PATH`?
I thought we were going with the second, but is `IMPORTED_LOCATION` better?

Reply:
From discussion this is just a `CMAKE` variable, any relation to an official variable is coincidental.
This is because we are looking for a `.so` file rather than a `cmake` hierarchy, so don't want to use `LIB_DIR` or `CMAKE_PREFIX_PATH` as we do in the Torch case which does have cmake capabilities.